### PR TITLE
[SC-210097] - Playwright session management

### PR DESCRIPTION
**Pull request by charles-testco**
### Summary:
- **What:** This PR introduces session management for playwright tests. It uses global setup to login once, save state into a file and attach this to all requests to going out. This way individual tests to not have to login.
The feature uses environment based config files, where credentials are stored. In the CI we will pass credentials through GITHUB secrets.
- **Ticket:** [sc-210687](https://app.shortcut.com/genepi/story/sc-210687)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [ ] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "devDependencies": {
+    "@playwright/test": "^1.23.3",
+    "babel-preset-env": "^1.7.0",
+    "lodash": "^4.17.21",
+    "parcel-bundler": "^1.12.5"
+  }
+}
```
Comment (phoenixAja) [inline package.json]: can you please merge this file and the package-lock file with the ones in `src/frontend`?

File: src/frontend/README-E2E.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## running tests
+To run Playwright tests you will need to specify the config file. We have created NPM scripts to make like a bit easier:
+
+`npm run pl:staging` - runs playwright tests against staging environment in headed browser
+`npm run pl:staging:headless` - same as above but headless
+`npm run pl:staging:debug` - runs Playwright in debug mode
+`npm run pl:report` opens test results report in the browser
+
+NOTES: 
+You need to be in the `src/frontend` to run these tests
```
Comment (phoenixAja) [inline src/frontend/README-E2E.md:14]: this is fine for now, i'll make a follow up ticket to add these commands to the makefile.

File: src/frontend/e2e/tests/home.spec.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  test("Should verify home page", async ({ page }, workerInfo) => {
+    const { baseURL } = workerInfo.config.projects[0].use;
+    await page.goto(`${baseURL}` as string);
+    await expect(page.locator(getByTestID("navbar-landing"))).toBeVisible();
+    await expect(
+      page.locator(getByTestID("navbar-sign-in-link"))
+    ).toBeVisible();
+    await expect(page.locator(getByTestID("logo"))).toBeVisible();
+    Object.keys(footer).forEach((key) => {
+      expect(page.locator(`a:has-text("${key}")`).first()).toHaveAttribute(
```
Comment (phoenixAja) [inline src/frontend/e2e/tests/home.spec.ts:23]: is this an ok selector to have?

**Review by phoenixAja — CHANGES_REQUESTED** [review]
please merge the package.json and package-lock.json with the ones in src/frontend

File: package.json (hunk 1 above)
Comment (ckmoga) [reply package.json]: Fixed

File: src/frontend/README-E2E.md (hunk 2 above)
Comment (ckmoga) [reply src/frontend/README-E2E.md:14]: Yes indeed, we need a separate ticket to configure this in makefile and CI.

File: src/frontend/e2e/tests/home.spec.ts (hunk 3 above)
Comment (ckmoga) [reply src/frontend/e2e/tests/home.spec.ts:23]: No they aren't but they are from PoC and have,'t been refactored. I will refactored them once I added test identifiers

**Review by phoenixAja — APPROVED** [review]
looks good! Thanks!

**Pull request merged**